src/picongpu/include/fields/MaxwellSolver/Yee/Curl.hpp

```diff
@@ -40,14 +38,10 @@ namespace yeeSolver
         template<class Memory >
         HDINLINE typename Memory::ValueType operator()(const Memory & mem) const
         {
-            Difference diff;
-            return float3_X(diff(mem, 1).z() - diff(mem, 2).y(),
-                               diff(mem, 2).x() - diff(mem, 0).z(),
-                               diff(mem, 0).y() - diff(mem, 1).x());
+            return float3_X(Difference::template GetDifference<1>()(mem).z() - Difference::template GetDifference<2>()(mem).y(),
```

Comment:
let us typedef a bit to create a sexy formula with linear operators `D_{x,y,z}` from it:

```c++
typedef typename Difference::GetDifference<0> Dx;
typedef typename Difference::GetDifference<1> Dy;
typedef typename Difference::GetDifference<2> Dz;

return float3_X( Dy()(mem).z() - Dz()(mem).y(),
                 Dz()(mem).x() - Dx()(mem).z(),
                 Dx()(mem).y() - Dy()(mem).x());
```